docs/api/v1/resourceset.md

```diff
@@ -374,6 +393,25 @@ spec:
 
 In the above example, the `ServiceAccount` resource is generated only for the `team1` tenant.
 
+#### Built-in input fields
+
+For every set of inputs, the reference of the object providing those inputs is injected
+by Flux Operator when templating the resources. The reference can be used in templates
+through the `inputs` object, through the following fields:
+
+- `inputs.provider.apiVersion`: The API version of the object providing the inputs.
+- `inputs.provider.kind`: The kind of the object providing the inputs.
+- `inputs.provider.name`: The name of the object providing the inputs.
+- `inputs.provider.namespace`: The namespace of the object providing the inputs.
```

Comment:
What happens if I have a RSET with in-line inputs like:

```yaml
spec:
  inputs:
   - provider: something
```

Reply:
The `provider` key is entirely replaced:

```go
	input["provider"] = map[string]any{
		"apiVersion": providerType.GroupVersion().String(),
		"kind":       providerType.Kind,
		"name":       provider.GetName(),
		"namespace":  provider.GetNamespace(),
	}
```

I added the following to the end of the docs section:

```
The `inputs.provider` field is always overwritten with the data above and cannot
be set by the user.
```